### PR TITLE
Mocking

### PR DESCRIPTION
**Pull request by marijnschilling**
I created a subclass of TeaPot to handle mocking. 

If you create a json file with the name of the endpoint of the api call it returns the mocked data. Otherwise it will return a `missingMockFile` error with the filename it expects, or when the JSON couldn't be parsed correctly it will return an `invalidMockFile` error with the filename.

### CODE REVIEW

**Comment by marijnschilling** [comment]
Too many merge conflicts in pod file :(

File: Teapot/MockTeapot.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    open var currentBundle: Bundle
+
+    public enum MockError: Error {
+        case missingMockFile(String)
+        case invalidMockFile(String)
+    }
+
+    public init(baseURL: URL, bundle: Bundle) {
+        currentBundle = bundle
```
Comment (elland) [inline Teapot/MockTeapot.swift]: Mind adding all the missing calls to `self`?

Comment (elland) [reply Teapot/MockTeapot.swift]: I'm running swiftformat first.

**Comment by marijnschilling** [comment]
What do we think @Elland? 😬

File: Teapot/MockTeapot.swift [hunk 2]
```diff
@@ -8,7 +9,12 @@ open class MockTeapot: Teapot {
         case missingMockFile(String)
         case invalidMockFile(String)
     }
-
+    
+    /// Initialiser.
+    ///
+    /// - Parameters:
+    ///   - baseURL: baseURL for the call, used to call the super init.
```
Comment (elland) [inline Teapot/MockTeapot.swift]: This isn't clear to me, also I think this is never actually used, if that's the case, can't we omit it from the init method here and just send a dummy value that works inside the method?

File: TeapotTests/MockTests.swift [hunk 3]
```diff
@@ -45,7 +45,7 @@ class MockTests: XCTestCase {
             case .failure(_, _, let error):
                 switch error {
                 case MockTeapot.MockError.invalidMockFile(let fileName):
-                    XCTAssertEqual(fileName, "invalid.json")
+                    XCTAssertEqual(fileName, "error: The data couldn’t be read because it isn’t in the correct format. in invalid.json")
```
Comment (elland) [inline TeapotTests/MockTests.swift]: The end of the sentence looks wrong to me. 😅

Comment (marijnschilling) [reply TeapotTests/MockTests.swift]: yup! Haha made it more clear now I think :)

File: Teapot/MockTeapot.swift [hunk 4]
```diff
@@ -13,12 +13,11 @@ open class MockTeapot: Teapot {
     /// Initialiser.
     ///
     /// - Parameters:
-    ///   - baseURL: baseURL for the call, used to call the super init.
     ///   - bundle: the bundle of your test target. When you add a json file with the name of the endpoint to your test target it will return this data.
-    public init(baseURL: URL, bundle: Bundle) {
```
Comment (elland) [inline Teapot/MockTeapot.swift]: 💥

File: Teapot/MockTeapot.swift [hunk 5]
```diff
@@ -44,7 +43,7 @@ open class MockTeapot: Teapot {
                     completion(nil, MockError.invalidMockFile("\(resource).json"))
                 }
             } catch let error {
-                completion(nil, MockError.invalidMockFile("error: \(error.localizedDescription) in \(resource).json"))
+                completion(nil, MockError.invalidMockFile("error: \(error.localizedDescription) In file: '\(resource).json'"))
```
Comment (elland) [inline Teapot/MockTeapot.swift]: 💪🏽

File: TeapotTests/MockTests.swift [hunk 6]
```diff
@@ -45,7 +45,7 @@ class MockTests: XCTestCase {
             case .failure(_, _, let error):
                 switch error {
                 case MockTeapot.MockError.invalidMockFile(let fileName):
-                    XCTAssertEqual(fileName, "error: The data couldn’t be read because it isn’t in the correct format. in invalid.json")
+                    XCTAssertEqual(fileName, "error: The data couldn’t be read because it isn’t in the correct format. In file: 'invalid.json'")
```
Comment (elland) [inline TeapotTests/MockTests.swift]: 🎸

**Review by elland — APPROVED** [review]
Ship it!

**Comment by marijnschilling** [comment]
🚢 😬

**Pull request merged**